### ISSUE TITLE
Add throttle option to register_batch_process

### ISSUE THREAD

**Issue by jjeaton** [reaktivstudios/locomotive#79]
Allow throttling of a batch process, for things like email notifications.

**Comment by zachwills** [comment]
@jjeaton did you have an idea of how you'd like to set this as a developer using the plugin? Was it something simple like `throttle => 60`, or did you have something more powerful in mind?

**Comment by jjeaton** [comment]
yep, basically a parameter for a value to pass to `usleep` in microseconds for it to pause in between processing

**Comment by Daronspence** [comment]
@jjeaton did you ever find a solution here? What about adding in the sleep function call to the callback function itself?

**Comment by jjeaton** [comment]
@Daronspence No solution as of yet, we haven't run into the use case ourselves yet. PRs always welcome :)
